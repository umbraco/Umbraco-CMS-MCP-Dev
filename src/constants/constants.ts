export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
