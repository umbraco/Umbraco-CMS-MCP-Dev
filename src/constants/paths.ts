import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');