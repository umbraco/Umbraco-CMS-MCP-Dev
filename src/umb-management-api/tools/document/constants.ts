export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}