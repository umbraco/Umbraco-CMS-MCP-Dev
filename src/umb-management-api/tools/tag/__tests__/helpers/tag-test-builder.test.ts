import { TagTestBuilder } from "./tag-test-builder.js";
import { DocumentTypeBuilder } from "../../../document-type/__tests__/helpers/document-type-builder.js";
import { DocumentTypeTestHelper } from "../../../document-type/__tests__/helpers/document-type-test-helper.js";
import { DocumentBuilder } from "../../../document/__tests__/helpers/document-builder.js";
import { DocumentTestHelper } from "../../../document/__tests__/helpers/document-test-helper.js";
import { TAG_DATA_TYPE_ID, BLANK_UUID } from "@/constants/constants.js";
import { jest } from "@jest/globals";

const TEST_DOCUMENT_NAME = "_Test Tag Builder Document";
const TEST_TAG_1 = "builder-test-tag";

describe("TagTestBuilder", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    // Clean up test documents
    await DocumentTestHelper.cleanup(TEST_DOCUMENT_NAME);
    await DocumentTypeTestHelper.cleanup(TEST_DOCUMENT_NAME);
  });

  it("should build tag search model with query and tagGroup", () => {
    const builder = new TagTestBuilder()
      .withQuery("test")
      .withTagGroup("default");

    const model = builder.build();

    expect(model).toMatchSnapshot();
  });


  it("should find specific tag", async () => {
    // Create test document with tags first
    const docTypeBuilder = await new DocumentTypeBuilder()
      .withName(TEST_DOCUMENT_NAME)
      .allowAsRoot(true)
      .withProperty("tag", "Tag", TAG_DATA_TYPE_ID)
      .create();

    const documentBuilder = await new DocumentBuilder()
      .withName(TEST_DOCUMENT_NAME)
      .withDocumentType(docTypeBuilder.getId())
      .withValue("tag", [TEST_TAG_1])
      .create();

    await documentBuilder.publish();

    const builder = new TagTestBuilder();
    const foundTag = await builder.findTag(TEST_TAG_1);

    // Normalize the tag ID for snapshot testing
    const normalizedTag = foundTag ? { ...foundTag, id: BLANK_UUID } : foundTag;
    expect(normalizedTag).toMatchSnapshot();
  });
  
});