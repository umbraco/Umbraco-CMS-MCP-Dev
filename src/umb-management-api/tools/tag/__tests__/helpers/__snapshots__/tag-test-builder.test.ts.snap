// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagTestBuilder should build tag search model with query and tagGroup 1`] = `
{
  "query": "test",
  "tagGroup": "default",
}
`;

exports[`TagTestBuilder should find specific tag 1`] = `
{
  "group": "default",
  "id": "00000000-0000-0000-0000-000000000000",
  "nodeCount": 1,
  "text": "builder-test-tag",
}
`;
